src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/NoMetadataFieldLayoutAlgorithm.cs

```diff
@@ -14,9 +14,6 @@ namespace Internal.Runtime.TypeLoader
     /// </summary>
     internal class NoMetadataFieldLayoutAlgorithm : FieldLayoutAlgorithm
     {
-#if SUPPORTS_NATIVE_METADATA_TYPE_LOADING
```

Comment:
There is also an instance of a related preprocessor: `SUPPORTS_NATIVE_METADATA_TYPE_LOADING_AND_SUPPORTS_TOKEN_BASED_DISPATCH_CELLS` (the one with the longest name in this repo 😅)
```
src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/CachedInterfaceDispatch.cs:#if SUPPORTS_NATIVE_METADATA_TYPE_LOADING_AND_SUPPORTS_TOKEN_BASED_DISPATCH_CELLS
src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeBuilderState.cs:                    // SUPPORTS_NATIVE_METADATA_TYPE_LOADING
```